src/NBitcoin/Coin.cs

```diff
@@ -772,22 +784,22 @@ public Script Redeem
             set;
         }
 
-        public override Script GetScriptCode()
+        public override Script GetScriptCode(Network network)
         {
             if(_OverrideScriptCode != null)
                 return _OverrideScriptCode;
             if(Redeem == null)
-                return base.GetScriptCode();
+                return base.GetScriptCode(network);
             else
-                return new ScriptCoin(this, Redeem).GetScriptCode();
+                return new ScriptCoin(this, Redeem).AssertCoherent(network).GetScriptCode(network);
```

Comment:
i've no idea what assert coherent is, but why have you added it here and below?

Reply:
Yes to avoid injecting the Network in the ScriptCoin class.
Ugly but its contained to NBitcoin anyone outside of NBitcoin will use the static `Create` method that takes network

Reply:
I could actually change this to use the static `Create` method

Reply:
Once again, I have no idea what you're talking about :) I still don't see why would you need to add this assert here.

Reply:
`AssertCoherent` was called inside the constructor of `ScriptCoin` so the options where eitehr pass it in the constructor or take the method out, I choose the second option.
But I will change this to use the `Create` method